### ISSUE TITLE
Mandatory and optional parameters

### ISSUE THREAD

**Issue by ankurhora** [InMobi/api-monetization#9]
Although it is mentioned in docs and also in the sample.php that only three parameters are needed for a successful hit 
1) Property ID
2) Device - Useragent 
3) Device - Ip
i tried doing that but failed no response but as soon as i added gpid in the api hit i start getting response. So my question is should not this also be mandatory ? or was i doing something wrong

**Comment by rishabh-chowdhary** [comment]
Yes, device ID(GPID & IDFA) are mandatory, I have to make this change in the codebase

**Comment by ankurhora** [comment]
Also, if requesting for android, setting gdip is mandatory and 
if requesting for ios, setting idfa is mandatory 
but there will be only one of them at a time. 
How are we handling that ? 
means: how should we call the device object ?
